### DISCUSSION TITLE
Support adding where condition to UniqueConstraint

### DISCUSSION THREAD

**Discussion by cankush625** [sqlalchemy/sqlalchemy#12044] [Usage Questions]
### Describe the use case

I want to add a where condition to UniqueConstraint so that the constraint can be applied to a selected set of records. I know this can be done using Index but I want to defer this constraint till the transaction is completed. I cannot defer the Index nor I can add where condition to UniqueConstraint.

Django seems to support adding where condition to UniqueConstraint. I would really appreciate having this feature in SQLAlchemy since it solves a common problem in efficient way.

### Databases / Backends / Drivers targeted

I'm looking this for PostgreSQL

### Example Use

In Django, this is supported like this:
```
models.UniqueConstraint(
    fields=["email", "tenant"],
    condition=Q(deleted_at=None),
    name="users_email_tenant_id_deleted_at_null_uniq",
    deferrable=Deferrable.DEFERRED,
),
```

Here, I'm applying the constraint on all the records which are not deleted and also deferring the check so that I can do bulk updates.

**Comment by CaselIT** [comment]
Hi,

I don't know what django does, but postgresql doesn't seem to support them: https://www.postgresql.org/docs/current/ddl-constraints.html
